### DISCUSSION TITLE
Tdx VmBoot(InvalidPayload) error on ubuntu Jamy

### DISCUSSION THREAD

**Discussion by oksanassss** [cloud-hypervisor/cloud-hypervisor#5739] [General]
when ı runned tdx example  , getting  this error :
```
Error booting VM: VmBoot(InvalidPayload)
``` 

cloned  source code from github. I builded  different version but get same error. 

versions :  v32.1.0 and  v34.0-57-gb1a1b2a6 

it's my  commands for run  machine with tdx : 

```
sudo ./cloud-hypervisor/target/release/cloud-hypervisor \
	    --firmware edk2/Build/IntelTdx/RELEASE_GCC5/FV/OVMF.fd \
	    --memory size=1G \
	    --kernel vmlinux.bin \
	    --cpus boot=1 \
	    --cmdline "console=ttyS0" \
	    --disk path=/home/oznfc/workdir/cloud-hypervisor/cloud-hypervisor/focal-server-cloudimg-amd64.raw --disk path=/tmp/ubuntu-cloudinit.img  \
	    --net "mac=12:34:56:78:90:ab,tap="
	    
```	    
 and it's my error : 
	    
 ```Error booting VM: VmBoot(InvalidPayload)```

**Comment by likebreath** [comment]
I am not sure which TDX example you are referring to, but it looks wrong to me the way how you use Cloud Hypervisor to launch TDs. 

Please refer to our latest TDX documentation for the latest examples and instructions [1]. I want to mention that you will need to properly setup your host environment, use a compatible guest kernel, guest image, etc. As mentioned in our documentation, one good reference about how to do the setup is using tdx tools [2].

[1] https://github.com/cloud-hypervisor/cloud-hypervisor/blob/main/docs/intel_tdx.md
[2] https://github.com/intel/tdx-tools
